### ISSUE TITLE
shell command ("bash -o pipefail") fails execution

### ISSUE THREAD

**Issue by kevinburke** [bazelbuild/rules_go#885]
I think this is a recent failure. I run Bazel in Travis CI, invoked with the following `travis.yml`

```yaml
go_import_path: github.com/kevinburke/nacl

language: go

os: linux
dist: trusty

sudo: required

before_script:
     - wget "https://storage.googleapis.com/bazel-apt/pool/jdk1.8/b/bazel/bazel_0.6.1_amd64.deb"
     - sudo dpkg --force-all -i bazel_0.6.1_amd64.deb
     - sudo apt-get install moreutils -y

script:
  - make ci

cache:
  directories:
    - $HOME/.bzrepos
```

Here's the contents of the `ci` Make target:

```
ci:
	bazel --batch --host_jvm_args=-Dbazel.DigestFunction=SHA1 test \
		--experimental_repository_cache="$$HOME/.bzrepos" \
		--spawn_strategy=remote \
		--remote_rest_cache=https://remote.rest.stackmachine.com/cache \
		--test_output=errors \
		--strategy=Javac=remote \
		--noshow_progress \
		--noshow_loading_progress \
		--features=race //... 2>&1 | ts '[%Y-%m-%d %H:%M:%.S]'
```

`ts` is a command to print timestamps, so I can see how long steps of the build take. However, this was causing the build to be incorrectly marked as passing, when it was actually failing. So I followed the advice here: https://stackoverflow.com/a/31605520/329700 to set `pipefail` in my Makefile:

```
SHELL := /bin/bash -o pipefail
```

I believe this is causing Bazel to fail. Here's the error message I get:

```
[2017-10-09 17:03:55.888958] Extracting Bazel installation...
[2017-10-09 17:04:20.966305] INFO: Analysed 25 targets (37 packages loaded).
[2017-10-09 17:04:20.967709] INFO: Found 11 targets and 14 test targets...
[2017-10-09 17:04:24.002989] ERROR: /home/travis/.cache/bazel/_bazel_travis/10212575102b38e278b4bbd4cd4fa80b/external/io_bazel_rules_go/go/tools/builders/BUILD:13:1: error executing shell command: '-o' failed (Exit 34). Note: Remote connection/protocol failed with: execution failed
[2017-10-09 17:04:24.191137] INFO: Elapsed time: 22.111s, Critical Path: 0.53s
[2017-10-09 17:04:24.197795] 
[2017-10-09 17:04:24.205235] Executed 0 out of 14 tests: 1 fails to build and 13 fail locally.
```

You can reproduce by forking the `update-deps` branch of github.com/kevinburke/nacl. I was also able to reproduce it locally on my Mac by checking out and running `make ci`.

**Comment by kevinburke** [comment]
hmm... I also get the error with rules_go 0.5.4 so maybe this is a problem with Bazel 0.6.1

**Comment by kevinburke** [comment]
Whoops, this was a combination of `remote.rest.stackmachine.com` no longer working as a Bazel cache, and a misleading error message. I reported the error message upstream as bazelbuild/bazel#3877.
